src/main/java/org/checkerframework/checker/builder/TypesafeBuilderAnnotatedTypeFactory.java

```diff
@@ -169,9 +184,30 @@ public boolean isSubtype(final AnnotationMirror subAnno, final AnnotationMirror
             } else if (AnnotationUtils.areSame(superAnno, BOTTOM)) {
                 return false;
             }
+
+            if (AnnotationUtils.areSame(superAnno, TOP)) {
+                return true;
+            } else if (AnnotationUtils.areSame(subAnno, TOP)) {
+                return false;
+            }
+
+            if (AnnotationUtils.areSameByClass(subAnno, CalledMethodsPredicate.class)) {
+                return false;
+            }
+
             Set<String> subVal = new LinkedHashSet<>(getValueOfAnnotationWithStringArgument(subAnno));
-            Set<String> superVal = new LinkedHashSet<>(getValueOfAnnotationWithStringArgument(superAnno));
-            return subVal.containsAll(superVal);
+
+            if (AnnotationUtils.areSameByClass(superAnno, CalledMethodsPredicate.class)) {
+                // superAnno is a CMP annotation, so we need to evaluate the predicate
+                String predicate = AnnotationUtils.getElementValue(superAnno, "value", String.class, false);
+                CalledMethodsPredicateEvaluator evaluator = new CalledMethodsPredicateEvaluator(subVal);
+                String result = evaluator.evaluate(predicate);
+                return Boolean.parseBoolean(result);
+            } else {
+                // superAnno is a CM annotation, so compare the sets
+                Set<String> superVal = new LinkedHashSet<>(getValueOfAnnotationWithStringArgument(superAnno));
```

Comment:
Is there a reason for `superVal`?  Doesn't `containsAll` take a `Collection` as an argument?